Makefile

```diff
@@ -60,6 +60,15 @@ i18n.preprocess: ## gather all display strings into a single file
 i18n.pre_validate: | i18n.extract i18n.preprocess
 	git diff --exit-code ./src/data/i18n/default/transifex_input.json
 
+tx_url1 = https://www.transifex.com/api/2/project/edx-platform/resource/studio-frontend/translation/en/strings/
+tx_url2 = https://www.transifex.com/api/2/project/edx-platform/resource/studio-frontend/source/
+push_translations: | i18n.extract
+	# Transifex is set up to watch transifex_input.json, but that filetype lacks comments
+	./node_modules/reactifex/bash_scripts/get_hashed_strings.sh $(tx_url1)
+	$$(npm bin)/reactifex ./src/data/i18n/default/src/components/ --comments
+	./node_modules/reactifex/bash_scripts/put_comments.sh $(tx_url2)
```

Comment:
See https://github.com/efischer19/reactifex/blob/master/README.md for details

Reply:
Cool package, but I can't help but think that JS dynamically creating a bash script seems code smelly. You mentioned that only curl was working for you... what was wrong with doing the request in Node instead?

Reply:
Agreed, it wasn't my first choice.

The problem came in when trying to get basic auth to succeed while handling 301 redirects ([Transifex docs)(https://docs.transifex.com/api/introduction#authentication)). I couldn't get [the default HTTPS Node library](https://nodejs.org/api/https.html) to do that correctly, so I defaulted back to the documented-by-Transifex approach of using curl.

It'd arguably be better to add a dependency on `[request](https://www.npmjs.com/package/request)` and use that; it just felt a little heavyweight compared to the dead-simple curls outlined in the transifex docs.

I just added some tests with an example of `put_comments.sh`, if seeing the results makes things any clearer. All the js does is match up message ids with Transifex-defined `string_hash` values to construct a URL. https://github.com/efischer19/reactifex/blob/master/bash_scripts/put_comments.sh

Reply:
as of the 1.0.2 release, the above linked script is now static while the *data it consumes* is dynamic, which seems way better.